MainWindow.xaml.cs

```diff
@@ -0,0 +1,2329 @@
+                int i = 0;
+                foreach (var row in text.Split("\n"))
+                {
+                    int j = 0;
+                    _globalnaListaZapisu.Add(new List<string>());
+                    foreach (var col in row.Trim().Split(_globalDelimiter))
+                    {
+                        _globalnaListaZapisu[i].Add(col);
+                        j++;
+                    }
+                    i++;
+                }
+
+                var JsonSave = JsonConvert.SerializeObject(_globalnaListaZapisu);
+                aktualnyStan.Write(JsonSave);
+            }
+            catch (Exception exception)
+            {
+                MessageBox.Show(exception.Message, "Wystąpił błąd");
+            }
+
+            aktualnyStan.Flush();
+            aktualnyStan.Close();
+        }
+
+        private void Button_Dodaj_Wiersz(object sender, RoutedEventArgs e)
+        {
+
+            bool tbDone = true;
+            bool format = false;
+            if (string.IsNullOrWhiteSpace(tbDodajWiersz.Text))
+            {
+                MessageBox.Show("Wartość nowego wiersza nie może być pusta", "Wystąpił błąd");
+                tbDone = false;
+            }
+
+            var delimiter = ",";
+
+            if (Path.GetFileName(_globalFilename) == @"australian.dat")
+                delimiter = " ";
+            if (Path.GetFileName(_globalFilename) == @"breast-cancer-wisconsin.data")
+                delimiter = ",";
+            if (Path.GetFileName(_globalFilename) == @"crx.data")
+                delimiter = ",";
+            if (_poNormalizacji)
+                delimiter = " ";
+
+            int liczbaKolumn = 0;
+            if (!string.IsNullOrWhiteSpace(_globalFilename))
+            {
+                var pierwszaLinia = File.ReadLines(_globalFilename).First();
+                liczbaKolumn = pierwszaLinia.Split(delimiter).Length;
+            }
+
+            if (tbDodajWiersz.Text.Split(delimiter).Length == liczbaKolumn)
+                format = true;
+            if (!format)
+                MessageBox.Show("Nie poprawny format nowego wiersza");
+            if (tbDone && format)
+            {
+                try
+                {
+                    using (StreamWriter w = File.AppendText(_globalFilename))
+                    {
+                        w.WriteLine(tbDodajWiersz.Text);
+                    }
+                    _globalSaveParagraph.Inlines.Add(tbDodajWiersz.Text);
+                    _globalDelimiter = delimiter;
+                }
+                catch (Exception exception)
+                {
+                    MessageBox.Show(exception.Message, "Wystąpił błąd");
+                    throw;
+                }
+                MessageBox.Show("Pomyślnie dodano nowy wiersz o wartości: " + tbDodajWiersz.Text);
+            }
+            if (!_poNormalizacji)
+                Button_Open(sender, e);
+            _dodanyWiersz++;
+        }
```

Comment:
to tez, zawsze mozna powiedziec zeby to dodali w pliku datasetu i masz funkcjonalnosc z glowy haha :D no ale jak jest i dziala to niech zostanie